### ISSUE TITLE
``is_real`` doesn't work on ``Add`` or ``Mul`` objects

### ISSUE THREAD

**Issue by Cryoris** [symengine/symengine#1922]
I'm not sure if this is intended or not, but it seems that the ``is_real`` attribute does not work on ``Add`` or ``Mul`` objects (or maybe any other than ``ComplexDouble``?). For example, in the following cases I did expect that symengine identifies the expression as non-real 
```python
from symengine import var

x = var("x")
ix = 1j * x

cdouble = ix.subs({x: 1})  # 0.0 + 1.0 * I --> plain ComplexDouble
print(cdouble.is_real)  # False, as expected

mul = ix.subs({x: 2})  # 2 * (0.0 + 1.0 * I) --> Mul object!
print(mul.is_real)  # None! Shouldn't this be False too?
```
The same happened for `Add` objects. Is this by design or would it be possible to make `is_real` work on these objects too? 🙂

**Comment by isuruf** [comment]
> Also sympy autosimplifies this to 2.0*I should symengine also do this?

Yes, it's a bug.

**Comment by isuruf** [comment]
This is already fixed in master branch.

**Comment by rikardn** [comment]
Ok, great. Then we this issue can be closed directly when #1923 is in.
